akka-javasdk/src/main/java/akka/javasdk/agent/ConversationMemory.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.agent;
+
+import akka.Done;
+import akka.javasdk.agent.ConversationMemory.History;
+import akka.javasdk.agent.ConversationMemory.Event;
+import akka.javasdk.annotations.TypeName;
+import akka.javasdk.eventsourcedentity.EventSourcedEntity;
+import akka.javasdk.annotations.ComponentId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import static akka.Done.done;
+
+/**
+ * ConversationMemory is an EventSourcedEntity that maintains a limited history of conversation
+ * messages in a FIFO (First In, First Out) manner.
+ * <p>
+ * The maximum number of entries in the history can be set dynamically with command setLimitedWindow.
```

Comment:
Something about not users not instantiating it themselves but calling it using component client for any direct interactions.

Alternatively, if the users should not use component client, is the implementation actually internal rather than public, and should go in impl?